File: backend/models/Service.js
import { DataTypes } from 'sequelize';
import { sequelize } from '../config/database.js';

const Service = sequelize.define('Service', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  title: {
    type: DataTypes.STRING,
    allowNull: false
  },
  category: {
    type: DataTypes.STRING,
    allowNull: false
  },
  description: {
    type: DataTypes.TEXT,
    allowNull: true,
    defaultValue: ''
  },
  price: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false
  },
  image: {
    type: DataTypes.STRING,
    allowNull: true,
    defaultValue: ''
  },
  rating: {
    type: DataTypes.DECIMAL(3, 1),
    allowNull: false,
    defaultValue: 0
  },
  reviews: {
    type: DataTypes.INTEGER,
    allowNull: false,
    defaultValue: 0
  },
  providerId: {
    type: DataTypes.INTEGER,
    allowNull: true,
    defaultValue: null,
    references: {
      model: 'users',
      key: 'id'
    }
  },
  isActive: {
    type: DataTypes.BOOLEAN,
    allowNull: false,
    defaultValue: true
  }
}, {
  tableName: 'services',
  timestamps: true,
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
});

export default Service;

File: backend_fastapi/models/service.py
from typing import Optional
from sqlmodel import SQLModel, Field, Relationship
# from .user import User

class ServiceBase(SQLModel):
    title: str
    category: str
    description: Optional[str] = ""
    price: float
    image: Optional[str] = ""
    rating: float = Field(default=0.0)
    reviews: int = Field(default=0)
    isActive: bool = Field(default=True)
    providerId: Optional[int] = Field(default=None, foreign_key="user.id")

class Service(ServiceBase, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    
    provider: Optional["User"] = Relationship(back_populates="services")
    bookings: list["Booking"] = Relationship(back_populates="service")

from .booking import Booking
from .user import User
